File: homework_iv/homework_iv_server/src/main/java/rs/webprogramming/homework_iv/main/controller/AssistantController.java
package rs.webprogramming.homework_iv.main.controller;

import lombok.SneakyThrows;
import lombok.extern.slf4j.Slf4j;
import rs.webprogramming.homework_iv.main.service.DatabaseService;
import rs.webprogramming.homework_iv.main.service.DefaultDatabaseService;

import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.BufferedReader;
import java.io.PrintWriter;

@Slf4j
@WebServlet("/assistant-to-grade")
public class AssistantController extends HttpServlet {

    private final DatabaseService databaseService = new DefaultDatabaseService();

    @SneakyThrows
    public void doPost(HttpServletRequest request, HttpServletResponse response) {
        PrintWriter out = response.getWriter();
        response.setContentType("application/json");
        response.setStatus(200);

        StringBuilder buffer = new StringBuilder();
        BufferedReader reader = request.getReader();
        String line;
        while ((line = reader.readLine()) != null) {
            buffer.append(line);
        }
        String payload = buffer.toString();

        databaseService.saveAssistant(payload, out);

        out.close();
    }

    @SneakyThrows
    public void doGet(HttpServletRequest request, HttpServletResponse response) {
        PrintWriter out = response.getWriter();
        response.setContentType("application/json");
        response.setStatus(200);
        databaseService.listAssistants(out);
        out.close();
    }

}

File: homework_vi/src/main/java/rs/webprogramming/homework_vi/controller/AssistantController.kt
package rs.webprogramming.homework_vi.controller

import rs.webprogramming.homework_vi.repository.AssistantRepository
import rs.webprogramming.homework_vi.repository.MySqlAssistantRepository
import rs.webprogramming.homework_vi.service.DefaultRestResponseService
import rs.webprogramming.homework_vi.service.RestResponseService
import rs.webprogramming.homework_vi.util.mapToAssistant
import java.io.BufferedReader
import javax.servlet.Servlet
import javax.servlet.annotation.WebServlet
import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@WebServlet("/grading")
class AssistantController: HttpServlet(), Servlet {

    private val restResponseService: RestResponseService = DefaultRestResponseService()

    override fun doPost(request: HttpServletRequest?, response: HttpServletResponse?) {
        val buffer = StringBuilder()
        val reader: BufferedReader = request?.reader!!
        var line: String?
        while (reader.readLine().also { line = it } != null) {
            buffer.append(line)
        }
        val payload = buffer.toString()

        val assistant = payload.mapToAssistant()

        if(assistant.score!! < 1 || assistant.score > 10) {
            restResponseService.invalidScore(response!!, assistant)
            return
        }

        restResponseService.saveAssistant(request, response!!, assistant)
    }

}